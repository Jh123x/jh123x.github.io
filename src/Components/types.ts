export interface InterestDescription {
    image: HTMLImageElement;
    title: string;
    paragraph?: string;
    href: string;
}

export type MenuItem = "Home" | "Resume" | "Portfolios" | "About";