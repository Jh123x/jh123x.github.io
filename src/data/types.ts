export interface PortfolioLink {
    link: string;
    tooltip?: string;
}

export interface WorkExperienceEntry {
    year: string;
    logo: string;
    title: string;
    subTitle: string;
    text: string;
    links?: LinkTypes;
}

export interface LinkTypes {
    github?: PortfolioLink;
    youtube?: PortfolioLink;
    google?: PortfolioLink;
    link?: PortfolioLink;
    link2?: PortfolioLink;
    docs?: PortfolioLink;
    feature_site?: PortfolioLink;
    project_showcase?: PortfolioLink;
    official?: PortfolioLink;
}

export interface PortfolioEntry {
    title: string;
    text: string;
    categories: string[];
    image: string;
    links: LinkTypes;
}

export interface SchoolExperienceEntry {
    year: string;
    logo: string;
    title: string;
    subTitle: string;
    text: string;
}

export interface Title {
    title: string;
    span: string;
}