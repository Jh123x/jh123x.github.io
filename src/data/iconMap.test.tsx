import * as React from "react";
import itemMap from "./iconMap";
import { IconTypes, LinkTypes } from "./types";
import { describe, expect, test } from "@jest/globals";

type testCase = [IconTypes, LinkTypes];

describe("itemMap", () => {
  const tests: testCase[] = [
    [
      IconTypes.github,
      { github: { link: "https://github.com", tooltip: "Github" } },
    ],
    [
      IconTypes.google,
      { google: { link: "https://google.com", tooltip: "Google" } },
    ],
    [
      IconTypes.apple,
      { apple: { link: "https://apple.com", tooltip: "Apple" } },
    ],
    [
      IconTypes.linkedin,
      { linkedin: { link: "https://linkedin.com", tooltip: "LinkedIn" } },
    ],
    [
      IconTypes.link,
      {
        link: [
          { link: "https://google.com", tooltip: "Link" },
          { link: "https://google.com", tooltip: "Link2" },
        ],
      },
    ],
    [
      IconTypes.youtube,
      { youtube: { link: "https://youtube.com", tooltip: "YouTube" } },
    ],
    [IconTypes.docs, { docs: { link: "https://docs.com", tooltip: "Docs" } }],
  ];

  for (const [icon, links] of tests) {
    test(`should render a ${icon} icon`, () => {
      const wrapper = <div>{itemMap(links)}</div>;
      expect(wrapper).toMatchSnapshot();
    });
  }
});
