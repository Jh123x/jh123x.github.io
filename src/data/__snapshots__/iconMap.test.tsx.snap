// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`itemMap should render a apple icon 1`] = `
<div>
  <ul>
    <li>
      <ForwardRef(Tooltip)
        title="Apple"
      >
        <ForwardRef(LinkComponent)
          href="https://apple.com"
        >
          <Memo />
        </ForwardRef(LinkComponent)>
      </ForwardRef(Tooltip)>
    </li>
  </ul>
</div>
`;

exports[`itemMap should render a docs icon 1`] = `
<div>
  <ul>
    <li>
      <ForwardRef(Tooltip)
        title="Docs"
      >
        <ForwardRef(LinkComponent)
          href="https://docs.com"
        >
          <Memo />
        </ForwardRef(LinkComponent)>
      </ForwardRef(Tooltip)>
    </li>
  </ul>
</div>
`;

exports[`itemMap should render a github icon 1`] = `
<div>
  <ul>
    <li>
      <ForwardRef(Tooltip)
        title="Github"
      >
        <ForwardRef(LinkComponent)
          href="https://github.com"
        >
          <Memo />
        </ForwardRef(LinkComponent)>
      </ForwardRef(Tooltip)>
    </li>
  </ul>
</div>
`;

exports[`itemMap should render a google icon 1`] = `
<div>
  <ul>
    <li>
      <ForwardRef(Tooltip)
        title="Google"
      >
        <ForwardRef(LinkComponent)
          href="https://google.com"
        >
          <Memo />
        </ForwardRef(LinkComponent)>
      </ForwardRef(Tooltip)>
    </li>
  </ul>
</div>
`;

exports[`itemMap should render a link icon 1`] = `
<div>
  <ul>
    <li>
      <ForwardRef(Tooltip)
        title=""
      >
        <ForwardRef(LinkComponent)
          href=""
        >
          <Memo />
        </ForwardRef(LinkComponent)>
      </ForwardRef(Tooltip)>
    </li>
  </ul>
</div>
`;

exports[`itemMap should render a linkedin icon 1`] = `
<div>
  <ul>
    <li>
      <ForwardRef(Tooltip)
        title="LinkedIn"
      >
        <ForwardRef(LinkComponent)
          href="https://linkedin.com"
        >
          <Memo />
        </ForwardRef(LinkComponent)>
      </ForwardRef(Tooltip)>
    </li>
  </ul>
</div>
`;

exports[`itemMap should render a youtube icon 1`] = `
<div>
  <ul>
    <li>
      <ForwardRef(Tooltip)
        title="YouTube"
      >
        <ForwardRef(LinkComponent)
          href="https://youtube.com"
        >
          <Memo />
        </ForwardRef(LinkComponent)>
      </ForwardRef(Tooltip)>
    </li>
  </ul>
</div>
`;
